Mathlib/MeasureTheory/Group/FundamentalDomain.lean

```diff
@@ -719,4 +727,281 @@ protected theorem fundamentalInterior : IsFundamentalDomain G (fundamentalInteri
 
 end IsFundamentalDomain
 
+section FundamentalDomainMeasure
+
+variable (G) [Group G] [MulAction G α] [MeasurableSpace α]
+  (μ : Measure α)
+
+local notation "α_mod_G" => MulAction.orbitRel G α
+
+local notation "π" => @Quotient.mk _ α_mod_G
+
+variable {G}
+
+@[to_additive addQuotientMeasure_apply]
+lemma quotientMeasure_apply (s : Set α)
+    {U : Set (Quotient α_mod_G)} (meas_U : MeasurableSet U)  :
+    (μ.restrict s).map π U = μ ((π ⁻¹' U) ∩ s) := by
+  rw [Measure.map_apply (f := π) (fun V hV ↦ measurableSet_quotient.mp hV) meas_U,
+    Measure.restrict_apply (t := (Quotient.mk α_mod_G ⁻¹' U)) (measurableSet_quotient.mp meas_U)]
+
+@[to_additive IsAddFundamentalDomain.addQuotientMeasure_invariant]
+lemma IsFundamentalDomain.quotientMeasure_invariant [Countable G] [MeasurableSpace G] {s t : Set α}
+    [SMulInvariantMeasure G α μ] [MeasurableSMul G α] (fund_dom_s : IsFundamentalDomain G s μ)
+    (fund_dom_t : IsFundamentalDomain G t μ) :
+    (μ.restrict s).map π = (μ.restrict t).map π := by
+  ext U meas_U
+  rw [quotientMeasure_apply μ s meas_U]
+  rw [quotientMeasure_apply μ t meas_U]
+  apply MeasureTheory.IsFundamentalDomain.measure_set_eq fund_dom_s fund_dom_t
+  · exact measurableSet_quotient.mp meas_U
+  · intro g
+    ext x
+    have : Quotient.mk α_mod_G (g • x) = Quotient.mk α_mod_G x
+    · apply Quotient.sound
+      use g
+    aesop
+
+end FundamentalDomainMeasure
+
+/-! ## `HasFundamentalDomain` typeclass
+
+We define `HasFundamentalDomain` in order to be able to define the `covolume` of a quotient of `α`
+by a group `G`, which under reasonable conditions does not depend on the choice of fundamental
+domain. Even though any "sensible" action should have a fundamental domain, this is a rather
+delicate question which was recently addressed by Misha Kapovich: https://arxiv.org/abs/2301.05325
+
+TODO: Formalize the existence of a Dirichlet domain as in Kapovich's paper.
+
+-/
+
+section HasFundamentalDomain
+
+/-- We say a quotient of `α` by `G` `HasAddFundamentalDomain` if there is a measurable set
+  `s` for which `IsAddFundamentalDomain G s` holds. -/
+class HasAddFundamentalDomain (G α : Type*) [Zero G] [VAdd G α] [MeasureSpace α] : Prop where
+  has_add_fundamental_domain_characterization : ∃ s : Set α, IsAddFundamentalDomain G s
```

Comment:
the name sounds much more Lean 3 than Lean 4. `ExistsIsAddFundamentalDomain` would be better.

Reply:
will do, thanks